### DISCUSSION TITLE
Weatherproof Enclosure for RPI+HAT

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#447] [Q&A] [answered]
I currently have Intellichem.  It's still dosing correctly, but apparently serial communication with it and EasyTouch has failed.  

ScreenLogic still controls pump and reports temps but I can no longer get pH or ORP.  (But, they do show up and are reasonable on the Intellichem display)

If I have to replace the Intellichem pcb, it will be the third time.  And, they are expensive!  So, I've decided to go the RPI + REM approach.  But, I want to keep the Intellichem until I'm up and running with RPI and REM.

What are people using for an enclosure for RPI+HAT?   And, what is used to egress cables for ORP and ph?  There's probably room to spare in the Intellichem box, but I want to keep it going (it is dosing correctly) until I get up and going with RPI + REM.   So, I need a weatherproof enclosure.

I first need to verify which end of the Intellichem to EasyTouch serial interface has failed.  First, I'll check connections and continuity of the wires.  But, after that, seems I'll need to drag an o-scope into the pool equipment area and use logic analyzer probes to see which end isn't working properly.

Any ideas for an easier way to diagnose which end of the serial interface is broken?

**Comment by rstrouse** [comment]
Actually, if you hook up REM and dashPanel you can get the messages being transmitted between EasyTouch and IntelliChem using messageManager.  The communications are always initiated by EasyTouch and IntelliChem simply responds.

EasyTouch sends a message and IntelliChem responds.  The entirety of the protocol goes like this.

ET Sends   IC Responds
210            18 --- All settings and state in IntelliChem are returned
146              1 ---  Sends setpoint, tank, TA, CH, and CYA to IntelliChem


Oh and I like the Altelix fiberglass reinforced boxes.

**Reply by davthomaspilot** [reply]
There are probably no messages being exchanged between EasyTouch and Intellichem--I can confirm that using messageManager as you suggested.

But, it could be due to either side, couldn't it?   Does messageManager actually monitor the interface between EasyTouch and Intellichem?  Or, just what EasyTouch "thinks" it sent and what it receives?

**Comment by ibehren1** [comment, marked as answer]
> What are people using for an enclosure for RPI+HAT? And, what is used to egress cables for ORP and ph? 

I have been using this enclosure for about a year now.  Clean and dry through rain and the winter.  I expected there may be some heat issues but have not had any (using RPI 3).

https://smile.amazon.com/gp/product/B01N4FSKZM/ref=ppx_yo_dt_b_search_asin_image?ie=UTF8&psc=1

I mounted to the house beside the Intellichem and used outdoor rated CAT 6 to connect to contacts in the Intellichem and over to my heater.

**Reply by rstrouse** [reply]
If it does have some heat issues you can simply add an enclosure vent near the top of the box.  The idea is to let the hot air out.

**Reply by davthomaspilot** [reply]
Yes, looks great!  You helped me when I was figuring out intellichem messages a couple of years ago.  I used dashPanel with nodeJS pool controller and I got familiar with the REM then.  



Pretty sure the issue I'm having is just wiring between the two units.  Visual inspection of the (not weatherproof) telephone wiring reveled a spot that looked mechanically damaged.  I flexed that a bit, and now things are working again.   Off to Lowes to look for outdoor rated Cat5 or Cat6 

 

Both enclosures referenced look good.   Thanks!  Ordered up enclosure ibehren1 suggested. 

How do you get cables in and out of the box and keeping things waterproof?  And, if I had holes for ventilation, doesn't that compromise weather proofing?

.

**Reply by ibehren1** [reply]
<img width="704" alt="image" src="https://user-images.githubusercontent.com/7774746/168817223-5a719670-fc19-4be1-809a-725d0c91aa77.png">
<img width="740" alt="image" src="https://user-images.githubusercontent.com/7774746/168817660-bc723cda-5903-4139-b389-dc1eb78d7cbb.png">

Enclosure has good foam passthru for wiring.  Like I said, clean, dry and critter free still after a year.

On the bottom of the Intellichem there are grommets for wiring passthru.
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/7774746/168817600-f1e15981-29d5-4f9e-9ed1-bc942affa647.png">

**Reply by rstrouse** [reply]
Holes to not create an issue if they have a vent cover over them.
https://www.hawkusa.com/manufacturers/bud/enclosures/ipv-1115?gclid=EAIaIQobChMI7JHshtvm9wIVRBmtBh0xKAz5EAQYASABEgJ6tPD_BwE

In fact you run greater risk of condensate with a totally sealed enclosure than you do with a vented one.  Condensate collects on the cooler items inside the box and if the moisture is sealed in, it will continue to, for lack of a better term, rain on the inside of the box.  You will find that most NEMA enclosures suitable for electronics are not totally sealed (you couldn't toss them in the pool).   Collected moisture can exit the bottom of the box.  In both examples above the bottom of the box is sloped slightly so no moisture can collect and drip out of the less than sealed bottom of the box.  But water cannot ingress from above or splash from below.

**Comment by rstrouse** [comment]
I just wanted to make sure you saw this. https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller

**Comment by davthomaspilot** [comment]
Great Stuff!  Thanks guys.

**Reply by davthomaspilot** [reply]
My Intellichem problem last spring was just wiring.  I've always planned to do the DIY-REM-Chem-Controller, but figured I'd wait until the Intellichem died again before diving it.

Well, now's the time.  The touch panel on the Intellichem no longer works.  And,  that's the only way I can do some things I need to do, like manual dose and change maximum dosage.

First, I just wanted to check to make sure REM and node-js poolController are still maintained.  Yes?

**Reply by tagyoureit** [reply]
Yes.  Still maintained.  And very stable.  Assuming you've seen the [REM Chem wiki](https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-REM-Chem-Controller)?

**Reply by davthomaspilot** [reply]
Yes--nicely documented!

Thanks for the quick replies.
